src/MonC/Semantics/SemanticAnalyzer.cs

```diff
@@ -48,8 +48,10 @@ public void Analyze(ParseModule headerModule, ParseModule newModule)
                 _functions.Add(externalFunction.Name, externalFunction);
             }
             foreach (FunctionDefinitionNode function in newModule.Functions) {
-                if (_functions.ContainsKey(function.Name)) {
-                    _errorsToProcess.Add(("Redefinition of function " + function.Name, function));
+                if (_functions.TryGetValue(function.Name, out FunctionDefinitionNode existingFunction)) {
+                    if (!ReferenceEquals(function, existingFunction)) {
```

Comment:
In what case would we want to have a function definition node instance added to the list multiple times? Checking equality here seems like it's covering up a bug.

Reply:
The first pass adds *all* module functions and enums to the header module ahead of time. By the time the analysis occurs, a redefinition is inevitable.

The best solution I can think of is changing this test to a simple multiple-definition within the header module. In fact, it would be most efficient to perform this check on the header module alone in between the parsing pass and analysis pass.